app/Support/Cache/CacheKey.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Support\Cache;
+
+class CacheKey
+{
+    public static function buildUserCanTicketCacheKey(string $userName): string
+    {
+        return self::buildNormalizedUserCacheKey($userName, "canTicket");
+    }
+
+    public static function buildUserCardDataCacheKey(string $userName): string
+    {
+        return self::buildNormalizedUserCacheKey($userName, "card-data");
+    }
+
+    /**
+     * @param string $userName the name of the user for which the cache key is being constructed
+     * @param int $gameID the target game ID for the unlocks data
+     * @param int $flags The type of achievement flag which should correspond to values in the `AchievementType` enum.
+     *                   3 for 'OfficialCore' (default), 5 for 'Unofficial'.
```

Comment:
deptrac got really unhappy when I tried to pull in `AchievementType` into this class. This also occurred further down for `RankType`.

Documentation like this felt like the best compromise to still preserve developer experience.

Reply:
I see, deptrac only allows for the `Site` layer to be used in `Support`, apart from that `Support` is assumed to be shared but knowing nothing about the other layers 🤔 

Works like this, too, for now. That achievement type flag is a bit of a mess throughout the code, literally. It could just be replaced with a boolean here imo and add a suffix if it's set.

Reply:
I've replaced this with a boolean as recommended in https://github.com/RetroAchievements/RAWeb/pull/1659/commits/01bbb1c2d5a090e2536286aaa68e04990aee8883.

This still leaves the implicit use of `RankType` on line 36, which has three potential states (hardcore, softcore, trueratio). Not sure if we want to do anything about that one or not.